### ISSUE TITLE
Using Helicopter or Motorbike 

### ISSUE THREAD

**Issue by monajalal** [ai-tor/DeepGTAV#12]
Hi Aitor,

I saw in one of the posts you mentioned only the Toyota Prius is supported in your code. How can I use helicopter or motorbike? Can you please provide a suggestion?

Also, this other question might seem rather weird, but can I have a control on the number of nearby cars/people? to make less/more traffic/crowd through your code or other possible tools you might be aware of?

Thanks a ton,
Mona

**Comment by ai-tor** [comment]
Hi Mona,

Currently using other types of vehicles is not supported, but technicaly it is possible and it would be a good idea to implement it in the future as I have seen interest in some papers I read, especially for helicopters.

The second question is not weird at all, I think it is important. It is also not supported for now, but probably it will be implemented in the new February release.

Cheers!

**Comment by monajalal** [comment]
For the first question can you guide which files I should change? The codebase (Visual Studio Solution) is quite big. If you could give some starting point that would be really great.

**Comment by ai-tor** [comment]
Lines 33 and 34 of _Scenario.cpp_ tell the script what vehicle to load, so in theory, if you change the hash string to let's say the hash of an helicopter you should get it :)

These hashes can be found in the Internet: https://www.google.es/search?q=GTAV+vehicle+hashes&ie=utf-8&oe=utf-8&client=firefox-b-ab&gfe_rd=cr&ei=0sCBWLrgJ4PY8gf0tZHYAw

**Comment by monajalal** [comment]
Vehicle Hash Codes (for those who might be interested):
```
Model Name = hex,uint32,int32 (description)
------------------------------------
adder = 0xB779A091,3078201489,-1216765807
airbus = 0x4C80EB0E,1283517198,1283517198 (Airport Bus)
airtug = 0x5D0AAC8F,1560980623,1560980623
akuma = 0x63ABADE7,1672195559,1672195559
ambulance = 0x45D56ADA,1171614426,1171614426
annihilator = 0x31F0B376,837858166,837858166
armytanker = 0xB8081009,3087536137,-1207431159
armytrailer = 0xA7FF33F5,2818520053,-1476447243 (Army Flatbed Trailer)
armytrailer2 = 0x9E6B14D6,2657817814,-1637149482 (Flatbed With Cutter Trailer)
asea = 0x94204D89,2485144969,-1809822327
asea2 = 0x9441D8D5,2487343317,-1807623979 (Snowy Asea)
asterope = 0x8E9254FB,2391954683,-1903012613
bagger = 0x806B9CC3,2154536131,-2140431165
baletrailer = 0xE82AE656,3895125590,-399841706
baller = 0xCFCA3668,3486135912,-808831384
baller2 = 0x08852855,142944341,142944341 (RangeRover Evoque)
banshee = 0xC1E908D2,3253274834,-1041692462
barracks = 0xCEEA3F4B,3471458123,-823509173 (Barracks With Backcover)
barracks2 = 0x4008EABB,1074326203,1074326203 (Barracks Semi)
bati = 0xF9300CC5,4180675781,-114291515
bati2 = 0xCADD5D2D,3403504941,-891462355 (Bati Livery)
benson = 0x7A61B330,2053223216,2053223216
bfinjection = 0x432AA566,1126868326,1126868326
biff = 0x32B91AE8,850991848,850991848
bison = 0xFEFD644F,4278019151,-16948145
bison2 = 0x7B8297C5,2072156101,2072156101 (Cowboy Construction Bison)
bison3 = 0x67B3F020,1739845664,1739845664 (Landscapeing Bison)
bjxl = 0x32B29A4B,850565707,850565707
blazer = 0x8125BCF9,2166734073,-2128233223
blazer2 = 0xFD231729,4246935337,-48031959 (Lifeguard Blazer)
blazer3 = 0xB44F0582,3025077634,-1269889662 (Trevor's Hotrod Blazer)
blimp = 0xF7004C86,4143991942,-150975354
blista = 0xEB70965F,3950024287,-344943009
bmx = 0x43779C54,1131912276,1131912276
boattrailer = 0x1F3D44B5,524108981,524108981 (Boat Trailer)
bobcatxl = 0x3FC5D440,1069929536,1069929536
bodhi2 = 0xAA699BB6,2859047862,-1435919434 (Trevor's Truck)
boxville = 0x898ECCEA,2307837162,-1987130134 (Water&Power Boxville)
boxville2 = 0xF21B33BE,4061868990,-233098306 (Postal Boxville)
boxville3 = 0x07405E08,121658888,121658888 (Humane Boxville)
buccaneer = 0xD756460C,3612755468,-682211828
buffalo = 0xEDD516C6,3990165190,-304802106
buffalo2 = 0x2BEC3CBE,736902334,736902334 (Franklin's Buffalo)
bulldozer = 0x7074F39D,1886712733,1886712733
bullet = 0x9AE6DDA1,2598821281,-1696146015
burrito = 0xAFBB2CA4,2948279460,-1346687836 (Cowboy Construction, Phartegas, Atomic, Water & Power Burrito)
burrito2 = 0xC9E8FF76,3387490166,-907477130 (Bugstars Burrito)
burrito3 = 0x98171BD3,2551651283,-1743316013 (No livery Burrito)
burrito4 = 0x353B561D,893081117,893081117 (Cowboy Construction Burrito)
burrito5 = 0x437CF2A0,1132262048,1132262048 (Snowy Burrito)
bus = 0xD577C962,3581397346,-713569950
buzzard = 0x2F03547B,788747387,788747387
buzzard2 = 0x2C75F0DD,745926877,745926877 (Gunless / Transport Buzzard)
cablecar = 0xC6C3242D,3334677549,-960289747
caddy = 0x44623884,1147287684,1147287684 (Prolaps Caddy)
caddy2 = 0xDFF0594C,3757070668,-537896628 (Old Caddy)
camper = 0x6FD95F68,1876516712,1876516712
carbonizzare = 0x7B8AB45F,2072687711,2072687711
carbonrs = 0x00ABB0C0,11251904,11251904
cargobob = 0xFCFCB68B,4244420235,-50547061
cargobob2 = 0x60A7EA10,1621617168,1621617168 (Medical Cargobob)
cargobob3 = 0x53174EEF,1394036463,1394036463 (Trevor's Cargobob)
cargoplane = 0x15F27762,368211810,368211810
cavalcade = 0x779F23AA,2006918058,2006918058 (GTA IV Cavalcade/Cadillac Escalade 2005)
cavalcade2 = 0xD0EB2BE5,3505073125,-789894171 (GTA V Cavalcade/Cadillac Escalade 2013)
cheetah = 0xB1D95DA0,2983812512,-1311154784
coach = 0x84718D34,2222034228,-2072933068 (Dashound)
cogcabrio = 0x13B57D8A,330661258,330661258
comet2 = 0xC1AE4D16,3249425686,-1045541610
coquette = 0x067BC037,108773431,108773431
cruiser = 0x1ABA13B5,448402357,448402357
crusader = 0x132D5A1A,321739290,321739290
cuban800 = 0xD9927FE3,3650256867,-644710429
cutter = 0xC3FBA120,3288047904,-1006919392
daemon = 0x77934CEE,2006142190,2006142190
dilettante = 0xBC993509,3164157193,-1130810103
dilettante2 = 0x64430650,1682114128,1682114128 (Merryweather Patrol Car)
dinghy = 0x3D961290,1033245328,1033245328
dinghy2 = 0x107F392C,276773164,276773164 (2-Seater)
dloader = 0x698521E3,1770332643,1770332643
docktrailer = 0x806EFBEE,2154757102,-2140210194 (Shipping Container Trailer)
docktug = 0xCB44B1CA,3410276810,-884690486
dominator = 0x04CE68AC,80636076,80636076
double = 0x9C669788,2623969160,-1670998136
dubsta = 0x462FE277,1177543287,1177543287
dubsta2 = 0xE882E5F6,3900892662,-394074634 (Blacked Out Dubsta)
dump = 0x810369E2,2164484578,-2130482718
dune = 0x9CF21E0F,2633113103,-1661854193
dune2 = 0x1FD824AF,534258863,534258863 (Spacedocker)
duster = 0x39D6779E,970356638,970356638
elegy2 = 0xDE3D9D22,3728579874,-566387422
emperor = 0xD7278283,3609690755,-685276541
emperor2 = 0x8FC3AADC,2411965148,-1883002148 (Rusty Emperor)
emperor3 = 0xB5FCF74E,3053254478,-1241712818 (Snowy Emperor)
entityxf = 0xB2FE5CF9,3003014393,-1291952903
exemplar = 0xFFB15B5E,4289813342,-5153954
f620 = 0xDCBCBE48,3703357000,-591610296
faggio2 = 0x0350D1AB,55628203,55628203
fbi = 0x432EA949,1127131465,1127131465 (FIB Buffalo)
fbi2 = 0x9DC66994,2647026068,-1647941228 (FIB Granger)
felon = 0xE8A8BDA8,3903372712,-391594584
felon2 = 0xFAAD85EE,4205676014,-89291282 (Felon GT Convertible)
feltzer2 = 0x8911B9F5,2299640309,-1995326987
firetruk = 0x73920F8E,1938952078,1938952078
fixter = 0xCE23D3BF,3458454463,-836512833
flatbed = 0x50B0215A,1353720154,1353720154
forklift = 0x58E49664,1491375716,1491375716
fq2 = 0xBC32A33B,3157435195,-1137532101
freight = 0x3D6AAA9B,1030400667,1030400667 (Freight Train)
freightcar = 0x0AFD22A6,184361638,184361638 (Train Well Car)
freightcont1 = 0x36DCFF98,920453016,920453016 (Train Container)
freightcont2 = 0x0E512E79,240201337,240201337 (Train Container Livery)
freightgrain = 0x264D9262,642617954,642617954(Train Boxcar)
freighttrailer = 0xD1ABB666,3517691494,-777275802 (Freight Train Flatbed)
frogger = 0x2C634FBD,744705981,744705981
frogger2 = 0x742E9AC0,1949211328,1949211328 (Trevor's Frogger)
fugitive = 0x71CB2FFB,1909141499,1909141499
fusilade = 0x1DC0BA53,499169875,499169875
futo = 0x7836CE2F,2016857647,2016857647
gauntlet = 0x94B395C5,2494797253,-1800170043
gburrito = 0x97FA4F36,2549763894,-1745203402
graintrailer = 0x3CC7F596,1019737494,1019737494
granger = 0x9628879C,2519238556,-1775728740
gresley = 0xA3FC0F4D,2751205197,-1543762099
habanero = 0x34B7390F,884422927,884422927
handler = 0x1A7FCEFA,444583674,444583674
hauler = 0x5A82F9AE,1518533038,1518533038
hexer = 0x11F76C14,301427732,301427732
hotknife = 0x0239E390,37348240,37348240
infernus = 0x18F25AC7,418536135,418536135
ingot = 0xB3206692,3005245074,-1289722222
intruder = 0x34DD8AA1,886934177,886934177
issi2 = 0xB9CB3B69,3117103977,-1177863319 (Issi Convertible)
jackal = 0xDAC67112,3670438162,-624529134
jb700 = 0x3EAB5555,1051415893,1051415893
jet = 0x3F119114,1058115860,1058115860
jetmax = 0x33581161,861409633,861409633
journey = 0xF8D48E7A,4174679674,-120287622
khamelion = 0x206D1B68,544021352,544021352
landstalker = 0x4BA4E8DC,1269098716,1269098716
lazer = 0xB39B0AE6,3013282534,-1281684762
lguard = 0x1BF8D381,469291905,469291905
luxor = 0x250B0C5E,621481054,621481054
mammatus = 0x97E55D11,2548391185,-1746576111
manana = 0x81634188,2170765704,-2124201592
marquis = 0xC1CE1183,3251507587,-1043459709
maverick = 0x9D0450CA,2634305738,-1660661558
mesa = 0x36848602,914654722,914654722
mesa2 = 0xD36A4B44,3546958660,-748008636 (Snowy Mesa)
mesa3 = 0x84F42E51,2230595153,-2064372143 (Merryweather Mesa)
metrotrain = 0x33C9E158,868868440,868868440
minivan = 0xED7EADA4,3984502180,-310465116
mixer = 0xD138A6BB,3510150843,-784816453
mixer2 = 0x1C534995,475220373,475220373 (Wheels On Back)
monroe = 0xE62B361B,3861591579,-433375717
mower = 0x6A4BD8F6,1783355638,1783355638
mule = 0x35ED670B,904750859,904750859
mule2 = 0xC1632BEB,3244501995,-1050465301 (Drop Down Trunk)
nemesis = 0xDA288376,3660088182,-634879114
ninef = 0x3D8FA25C,1032823388,1032823388
ninef2 = 0xA8E38B01,2833484545,-1461482751 (Ninef Convertible)
oracle = 0x506434F6,1348744438,1348744438 (GTA IV Oracle)
oracle2 = 0xE18195B2,3783366066,-511601230 (Oracle XS)
packer = 0x21EEE87D,569305213,569305213
patriot = 0xCFCFEB3B,3486509883,-808457413
pbus = 0x885F3671,2287941233,-2007026063 (Prison Bus)
pcj = 0xC9CEAF06,3385765638,-909201658
penumbra = 0xE9805550,3917501776,-377465520
peyote = 0x6D19CCBC,1830407356,1830407356
phantom = 0x809AA4CB,2157618379,-2137348917
phoenix = 0x831A21D5,2199527893,-2095439403
picador = 0x59E0FBF3,1507916787,1507916787
police = 0x79FBB0C5,2046537925,2046537925 (Police Stanier)
police2 = 0x9F05F101,2667966721,-1627000575 (Police Buffalo)
police3 = 0x71FA16EA,1912215274,1912215274 (Police Interceptor)
police4 = 0x8A63C7B9,2321795001,-1973172295 (Undercover Police Stanier)
policeb = 0xFDEFAEC3,4260343491,-34623805 (Police Bike)
policeold1 = 0xA46462F7,2758042359,-1536924937 (Snowy Police Rancher)
policeold2 = 0x95F4C618,2515846680,-1779120616 (Snowy Police Esperanto)
policet = 0x1B38E955,456714581,456714581 (Police Transport Van)
polmav = 0x1517D4D9,353883353,353883353
pony = 0xF8DE29A8,4175309224,-119658072 (Sunset Bleach, Sprunk, Postal Pony)
pony2 = 0x38408341,943752001,943752001 (Weed Van)
pounder = 0x7DE35E7D,2112052861,2112052861
prairie = 0xA988D3A2,2844316578,-1450650718
pranger = 0x2C33B46E,741586030,741586030
predator = 0xE2E7D4AB,3806844075,-488123221
premier = 0x8FB66F9B,2411098011,-1883869285
primo = 0xBB6B404F,3144368207,-1150599089
proptrailer = 0x153E1B0A,356391690,356391690 (Mobile Home Trailer)
radi = 0x9D96B45B,2643899483,-1651067813
raketrailer = 0x174CB172,390902130,390902130 (Farm Cultivator)
rancherxl = 0x6210CBB0,1645267888,1645267888
rancherxl2 = 0x7341576B,1933662059,1933662059 (Snowy Rancher)
rapidgt = 0x8CB29A14,2360515092,-1934452204
rapidgt2 = 0x679450AF,1737773231,1737773231 (Rapid GT Convertible)
ratloader = 0xD83C13CE,3627815886,-667151410
rebel = 0xB802DD46,3087195462,-1207771834 (Rusty Rebel)
rebel2 = 0x8612B64B,2249373259,-2045594037 (Clean Rebel)
regina = 0xFF22D208,4280472072,-14495224
rentalbus = 0xBE819C63,3196165219,-1098802077
rhino = 0x2EA68690,782665360,782665360
riot = 0xB822A1AA,3089277354,-1205689942
ripley = 0xCD935EF9,3448987385,-845979911
rocoto = 0x7F5C91F1,2136773105,2136773105
romero = 0x2560B2FC,627094268,627094268
rubble = 0x9A5B1DCC,2589662668,-1705304628
ruffian = 0xCABD11E8,3401388520,-893578776
ruiner = 0xF26CEFF9,4067225593,-227741703
rumpo = 0x4543B74D,1162065741,1162065741 (Weazel News Rumpo)
rumpo2 = 0x961AFEF7,2518351607,-1776615689 (Deludamol Rumpo)
sabregt = 0x9B909C94,2609945748,-1685021548
sadler = 0xDC434E51,3695398481,-599568815
sadler2 = 0x2BC345D1,734217681,734217681 (Snowy Sadler)
sanchez = 0x2EF89E46,788045382,788045382 (Sanchez Livery)
sanchez2 = 0xA960B13E,2841686334,-1453280962 (Sanchez Paint)
sandking = 0xB9210FD0,3105951696,-1189015600 (Sandking 4-Seater)
sandking2 = 0x3AF8C345,989381445,989381445 (Sandking 2-Seater)
schafter2 = 0xB52B5113,3039514899,-1255452397
schwarzer = 0xD37B7976,3548084598,-746882698
scorcher = 0xF4E1AA15,4108429845,-186537451
scrap = 0x9A9FD3DF,2594165727,-1700801569
seashark = 0xC2974024,3264692260,-1030275036 (Speedophile Seashark)
seashark2 = 0xDB4388E4,3678636260,-616331036 (Lifeguard Seashark)
seminole = 0x48CECED3,1221512915,1221512915
sentinel = 0x50732C82,1349725314,1349725314 (Sentinel XS)
sentinel2 = 0x3412AE2D,873639469,873639469 (Sentinel Convertible)
serrano = 0x4FB1A214,1337041428,1337041428
shamal = 0xB79C1BF5,3080461301,-1214505995
sheriff = 0x9BAA707C,2611638396,-1683328900 (Sheriff Stanier)
sheriff2 = 0x72935408,1922257928,1922257928 (Sheriff Granger)
skylift = 0x3E48BF23,1044954915,1044954915
speedo = 0xCFB3870C,3484649228,-810318068
speedo2 = 0x2B6DC64A,728614474,728614474 (Clown Van)
squalo = 0x17DF5EC2,400514754,400514754
stanier = 0xA7EDE74D,2817386317,-1477580979
stinger = 0x5C23AF9B,1545842587,1545842587
stingergt = 0x82E499FA,2196019706,-2098947590
stockade = 0x6827CF72,1747439474,1747439474
stockade3 = 0xF337AB36,4080511798,-214455498 (Snowy Stockade)
stratum = 0x66B4FC45,1723137093,1723137093
stretch = 0x8B13F083,2333339779,-1961627517
stunt = 0x81794C70,2172210288,-2122757008
submersible = 0x2DFF622F,771711535,771711535
sultan = 0x39DA2754,970598228,970598228
suntrap = 0xEF2295C9,4012021193,-282946103
superd = 0x42F2ED16,1123216662,1123216662
surano = 0x16E478C1,384071873,384071873
surfer = 0x29B0DA97,699456151,699456151
surfer2 = 0xB1D80E06,2983726598,-1311240698 (Rusty Surfer)
surge = 0x8F0E3594,2400073108,-1894894188
taco = 0x744CA80D,1951180813,1951180813
tailgater = 0xC3DDFDCE,3286105550,-1008861746
tanker = 0xD46F4737,3564062519,-730904777
tankercar = 0x22EDDC30,586013744,586013744 (Train Fuel Tank Car)
taxi = 0xC703DB5F,3338918751,-956048545
tiptruck = 0x02E19879,48339065,48339065 (6-Wheeler)
tiptruck2 = 0xC7824E5E,3347205726,-947761570 (10-Wheeler)
titan = 0x761E2AD3,1981688531,1981688531
tornado = 0x1BB290BC,464687292,464687292
tornado2 = 0x5B42A5C4,1531094468,1531094468 (Tornado Convertible)
tornado3 = 0x690A4153,1762279763,1762279763 (Rusty Tornado)
tornado4 = 0x86CF7CDD,2261744861,-2033222435 (Tornado With Guitar/Mariachi Car)
tourbus = 0x73B1C3CB,1941029835,1941029835
towtruck = 0xB12314E0,2971866336,-1323100960 (Large Towtruck)
towtruck2 = 0xE5A2D6C6,3852654278,-442313018 (Small Towtruck)
tr2 = 0x7BE032C6,2078290630,2078290630 (Car Carrier Trailer)
tr3 = 0x6A59902D,1784254509,1784254509 (Marquis Trailer)
tr4 = 0x7CAB34D0,2091594960,2091594960 (Super Car Carrier Trailer)
tractor = 0x61D6BA8C,1641462412,1641462412 (Rusty Tractor)
tractor2 = 0x843B73DE,2218488798,-2076478498 (Farm Tractor)
tractor3 = 0x562A97BD,1445631933,1445631933 (Snowy Tractor)
trailerlogs = 0x782A236D,2016027501,2016027501 (Log Trailer)
trailers = 0xCBB2BE0E,3417488910,-877478386 (Metal/Tarp Covered Trailer)
trailers2 = 0xA1DA3C91,2715434129,-1579533167 (Up & Atom, Cluckinbell, Piswasser Trailer)
trailers3 = 0x8548036D,2236089197,-2058878099 (Biggoods Trailer)
trailersmall = 0x2A72BEAB,712162987,712162987 (Small Construction Trailer)
trash = 0x72435A19,1917016601,1917016601
trflat = 0xAF62F6B2,2942498482,-1352468814 (Flatbed Trailer)
tribike = 0x4339CD69,1127861609,1127861609 (Green Whippet Race Bike)
tribike2 = 0xB67597EC,3061159916,-1233807380 (Red Endurex Race Bike)
tribike3 = 0xE823FB48,3894672200,-400295096 (Blue Tri-Cycles Race Bike)
tropic = 0x1149422F,290013743,290013743
tvtrailer = 0x967620BE,2524324030,-1770643266 (Fame or Shame Trailer)
utillitruck = 0x1ED0A534,516990260,516990260 (Building & Renovation Basket Truck)
utillitruck2 = 0x34E6BF6B,887537515,887537515 (Landscape, Gas, Building & Renovation Utillitruck)
utillitruck3 = 0x7F2153DF,2132890591,2132890591 (Landscape Utility Pick-up Truck)
vacca = 0x142E0DC3,338562499,338562499
vader = 0xF79A00F7,4154065143,-140902153
velum = 0x9C429B6A,2621610858,-1673356438
vigero = 0xCEC6B9B7,3469130167,-825837129
voltic = 0x9F4B77BE,2672523198,-1622444098
voodoo2 = 0x1F3766E3,523724515,523724515 (Rusty Voodoo)
washington = 0x69F06B57,1777363799,1777363799
youga = 0x03E5F6B8,65402552,65402552
zion = 0xBD1B39C3,3172678083,-1122289213 (Zion XS)
zion2 = 0xB8E2AE18,3101863448,-1193103848 (Zion Convertible)
ztype = 0x2D3BD401,758895617,758895617

Dlc Vehicles:
------------------------------------
bifta = 0xEB298297,3945366167,-349601129
kalahari = 0x05852838,92612664,92612664
paradise = 0x58B3979C,1488164764,1488164764
speeder = 0x0DC60D2B,231083307,231083307
btype = 0x06FF6914,117401876,117401876 (Roosevelt)
jester = 0xB2A716A3,2997294755,-1297672541
turismor = 0x185484E1,408192225,408192225
alpha = 0x2DB8D1AA,767087018,767087018
vestra = 0x4FF77E37,1341619767,1341619767
massacro = 0xF77ADE32,4152024626,-142942670
zentorno = 0xAC5DF515,2891838741,-1403128555
huntley = 0x1D06D681,486987393,486987393
thrust = 0x6D6F8F43,1836027715,1836027715
rhapsody = 0x322CF98F,841808271,841808271
warrener = 0x51D83328,1373123368,1373123368
blade = 0xB820ED5E,3089165662,-1205801634
glendale = 0x047A6BC1,75131841,75131841
panto = 0xE644E480,3863274624,-431692672
dubsta3 = 0xB6410173,3057713523,-1237253773 (Dubsta 6x6)
pigalle = 0x404B6381,1078682497,1078682497
monster = 0xCD93A7DB,3449006043,-845961253 (The Liberator)
sovereign = 0x2C509634,743478836,743478836
besra = 0x6CBD1D6D,1824333165,1824333165
miljet = 0x09D80F93,165154707,165154707
coquette2 = 0x3C4E2113,1011753235,1011753235 (Coquette Classic)
swift = 0xEBC24DF2,3955379698,-339587598
innovation = 0xF683EACA,4135840458,-159126838
hakuchou = 0x4B6C568A,1265391242,1265391242
furoregt = 0xBF1691E0,3205927392,-1089039904
jester2 = 0xBE0E6126,3188613414,-1106353882 (Rester Race)
massacro2 = 0xDA5819A3,3663206819,-631760477 (Massacaro Race)
ratloader2 = 0xDCE1D9F7,3705788919,-589178377 (Rat Truck)
slamvan = 0x2B7F9DE3,729783779,729783779
mule3 = 0x85A5B471,2242229361,-2052737935 (No Livery)
velum2 = 0x403820E8,1077420264,1077420264 (5-Seater)
tanker2 = 0x74998082,1956216962,1956216962 (No Livery)
casco = 0x3822BDFE,941800958,941800958
boxville4 = 0x1A79847A,444171386,444171386 (Post OP)
hydra = 0x39D6E83F,970385471,970385471
insurgent = 0x9114EADA,2434067162,-1860900134 (Mounted Gun)
insurgent2 = 0x7B7E56F0,2071877360,2071877360 (Transport)
gburrito2 = 0x11AA0E14,296357396,296357396 (No Livery)
technical = 0x83051506,2198148358,-2096818938
dinghy3 = 0x1E5E54EA,509498602,509498602 (New Map On Dash)
savage = 0xFB133A17,4212341271,-82626025
enduro = 0x6882FA73,1753414259,1753414259
guardian = 0x825A9F4C,2186977100,-2107990196
lectro = 0x26321E67,640818791,640818791
kuruma = 0xAE2BFE94,2922118804,-1372848492
kuruma2 = 0x187D938D,410882957,410882957 (Armored)
trash2 = 0xB527915C,3039269212,-1255698084 (Rusty Back)
barracks3 = 0x2592B5CF,630371791,630371791 (Dark Camo and New Cover)
valkyrie = 0xA09E15FD,2694714877,-1600252419
slamvan2 = 0x31ADBBFC,833469436,833469436 (Lost Slamvan)
swift2 = 0x4019CB4C,1075432268,1075432268 (Gold)
luxor2 = 0xB79F589E,3080673438,-1214293858 (Gold)
feltzer3 = 0xA29D6D10,2728226064,-1566741232 (Stirling GT)
osiris = 0x767164D6,1987142870,1987142870
virgo = 0xE2504942,3796912450,-498054846
windsor = 0x5E4327C8,1581459400,1581459400
coquette3 = 0x2EC385FE,784565758,784565758 (Blackfin)
vindicator = 0xAF599F01,2941886209,-1353081087
t20 = 0x6322B39A,1663218586,1663218586
brawler = 0xA7CE1BC5,2815302597,-1479664699
toro = 0x3FD5AA2F,1070967343,1070967343
chino = 0x14D69010,349605904,349605904

Next Gen Vehicles:
------------------------------------
submersible2 = 0xC07107EE,3228633070,-1066334226 (Kraken)
dukes = 0x2B26F456,723973206,723973206
dukes2 = 0xEC8F7094,3968823444,-326143852 (Duke O'Death)
buffalo3 = 0x0E2C013E,237764926,237764926 (Sprunk Buffalo)
dominator2 = 0xC96B73D9,3379262425,-915704871 (Pisswasser Dominator)
dodo = 0xCA495705,3393804037,-901163259
marshall = 0x49863E9C,1233534620,1233534620
blimp2 = 0xDB6B4924,3681241380,-613725916 (Xero Blimp)
gauntlet2 = 0x14D22159,349315417,349315417 (Redwood Gauntlet)
stalion = 0x72A4C31E,1923400478,1923400478
stalion2 = 0xE80F67EE,3893323758,-401643538 (Burger Shot Stallion)
blista2 = 0x3DEE5EDA,1039032026,1039032026 (Blista Compact)
blista3 = 0xDCBC1C3B,3703315515,-591651781 (Go Go Monkey Blista)
```

**Comment by monajalal** [comment]
So I selected besra as my vehicle (kind of an aircraft), and in mode=0 the game AI moves the aircraft but it doesn't seem very smart like it crashes stuff, it is not soaring high. I wonder if you think your code can be applicable to other vehicles than like a car (it worked perfectly for a bobcatxl when I tried it). I changed the x/y position but no major change happened and the game AI seems very off). Any point or suggestion is really appreciated.
here is a clip from the AI playing the game for a besra:
https://drive.google.com/file/d/0B7j2NoupYo5gU05iNTBBMnV5UWs/view?usp=sharing
